### PR TITLE
getMessageAtIndex should actually return the value in the streamMessage for compatibility

### PR DESCRIPTION
**Pull request by navina**
Fixing a bug from https://github.com/apache/pinot/pull/10995

### CODE REVIEW

**Comment by navina** [comment]
@KKcorps  : review pls!

@Jackie-Jiang should I clean up this deprecated spi- method (`MessageBatch#getMessageAtIndex`) before the next release?

File: pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarMessageBatch.java [hunk 1]
```diff
@@ -49,11 +49,16 @@ public int getMessageCount() {
...
+  @Deprecated
   @Override
   public PulsarStreamMessage getMessageAtIndex(int index) {
     return _messageList.get(index);
   }
 
+  @Override
+  public byte[] getMessageBytesAtIndex(int index) {
+    return getMessageAtIndex(index).getValue();
+  }
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarMessageBatch.java:61]: (minor) Add an empty line

File: pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarMessageBatch.java [hunk 2]
```diff
@@ -49,11 +49,16 @@ public int getMessageCount() {
...
 
+  @Deprecated
   @Override
   public PulsarStreamMessage getMessageAtIndex(int index) {
     return _messageList.get(index);
   }
 
+  @Override
+  public byte[] getMessageBytesAtIndex(int index) {
+    return getMessageAtIndex(index).getValue();
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarMessageBatch.java:60]: (minor) Let's not use the deprecated method
```suggestion
    return _messageList.get(index).getValue();
```

**Review by Jackie-Jiang — APPROVED** [review]
The fix looks good to me.

Not introduced in this PR, but why do we prefer reading the bytes instead of directly reading the message?

**Comment by navina** [comment]
> Not introduced in this PR, but why do we prefer reading the bytes instead of directly reading the message?

The original change I introduced in #9224 was to cleanly decouple the interfaces for "fetching" a record from stream and "decoding" the payload. But that broke LinkedIn's build which was using a custom implementation of stream consumer and doesn't provide the capability to fetch raw bytes from kafka stream. So, it was fixed forward in #9544 .

**Pull request merged**